community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/sampling/IndexSamplingJobTracker.java

```diff
@@ -49,6 +51,7 @@ public boolean canExecuteMoreSamplingJobs()
         lock.lock();
         try
         {
+            assertNotStopped();
```

Comment:
I've made this assertion here to keep the implementation of `waitUntilCanExecuteMoreSamplingJobs()` unchanged.
But I agree throwing is counter intuitive here. Will fix.